mlkem/indcpa.c

```diff
@@ -155,34 +157,69 @@ static unsigned int rej_uniform(int16_t *r,
 #define GEN_MATRIX_NBLOCKS ((12*KYBER_N/8*(1 << 12)/KYBER_Q + XOF_BLOCKBYTES)/XOF_BLOCKBYTES)
 // Not static for benchmarking
 void gen_matrix(polyvec *a, const uint8_t seed[KYBER_SYMBYTES], int transposed) {
-    unsigned int ctr, i, j, k;
-    unsigned int buflen, off;
-    uint8_t buf[GEN_MATRIX_NBLOCKS * XOF_BLOCKBYTES + 2];
+    unsigned int ctr, i, j;
+    unsigned int buflen;
+    uint8_t bufx[KECCAK_WAY][GEN_MATRIX_NBLOCKS *XOF_BLOCKBYTES + 2];
+    uint8_t *buf = NULL;
+    int16_t *vec = NULL;
     xof_state state;
 
-    for (i = 0; i < KYBER_K; i++) {
-        for (j = 0; j < KYBER_K; j++) {
+    keccakx kx;
+    uint8_t seedxy[KECCAK_WAY][KYBER_SYMBYTES + 2];
```

Comment:
The size of `KYBER_SYMBTES + 2` is `34`, which creates unaligned load from stack for each lane. 
I suggest make it to `KYBER_SYMBYTES + 16`. Wdyt?

Reply:
Sure, but please add a comment.

Reply:
Comment added.